src/components/SingleDatePicker.jsx

```diff
@@ -84,23 +90,46 @@ const defaultProps = {
 export default class SingleDatePicker extends React.Component {
   constructor(props) {
     super(props);
+
+    this.isTouchDevice = false;
+    this.today = moment();
+
+    this.modifiers = {
+      today: day => this.isToday(day),
+      blocked: day => this.isBlocked(day),
+      // TODO: update these if they change
+      'blocked-calendar': day => props.isDayBlocked(day),
+      'blocked-out-of-range': day => props.isOutsideRange(day),
+      'highlighted-calendar': day => props.isDayHighlighted(day),
+      valid: day => !this.isBlocked(day),
+      hovered: day => this.isHovered(day),
+      selected: day => this.isSelected(day),
+    };
+
+    const initialVisibleMonthThunk = props.initialVisibleMonth || (() => (props.date || moment()));
```

Comment:
this could be more efficient with `props.initialVisibleMonth || (props.date ? () => props.date : () => moment())` (and even more so if the moment thunk was cached at module level instead of recreated every render)